lib/reply.js

```diff
@@ -139,9 +139,14 @@ function onSendEnd (reply, payload) {
     flatstr(payload)
   }
 
+  if (typeof payload !== 'string' && !Buffer.isBuffer(payload)) {
+    throw new Error(`Serializer for Content-Type '${contentType}' can not handle payload of type '${typeof payload}'`)
```

Comment:
If I saw this error while developing my application, I wouldn't understand what went wrong. What if I didn't set a custom serializer?

I think there should be 2 places to throw to give a better message:
1. After the custom serializer, so the message can be something like: `Serializer for Content-Type '${contentType}' returned invalid payload of type '${typeof payload}'. Expected a string or Buffer.`
2. If no serializer is used, so the message could be: `Attempted to send payload of invalid type '${typeof payload}' without serialization`

(I'm assuming the JSON serializer will always either return a valid payload or throw an error. Let me know if this assumption is incorrect.)

Reply:
Much better :thumbsup:

Reply:
Fixed